octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py

```diff
@@ -0,0 +1,232 @@
+            cryptofeed_constants.L3_BOOK: cryptofeed_callbacks.BookCallback(self.book),
+            cryptofeed_constants.FUNDING: cryptofeed_callbacks.FundingCallback(self.funding),
+            cryptofeed_constants.LIQUIDATIONS: cryptofeed_callbacks.LiquidationCallback(self.liquidations),
+            cryptofeed_constants.BOOK_DELTA: cryptofeed_callbacks.BookUpdateCallback(self.delta),
+            cryptofeed_constants.VOLUME: cryptofeed_callbacks.VolumeCallback(self.volume),
+            cryptofeed_constants.OPEN_INTEREST: cryptofeed_callbacks.OpenInterestCallback(self.open_interest),
+            cryptofeed_constants.FUTURES_INDEX: cryptofeed_callbacks.FuturesIndexCallback(self.futures_index),
+            cryptofeed_constants.MARKET_INFO: cryptofeed_callbacks.MarketInfoCallback(self.market_info),
+            cryptofeed_constants.TRANSACTIONS: cryptofeed_callbacks.TransactionsCallback(self.transactions),
+        }
+
+    @classmethod
+    def get_feed_name(cls):
+        raise NotImplementedError("get_feed_name not implemented")
+
+    def subscribe_candle_feed(self, exchange_symbols):
+        candle_callback = self.callback_by_feed[self.EXCHANGE_FEEDS[Feeds.CANDLE]]
+
+        for time_frame in self.time_frames:
+            self.client.add_feed(self.get_feed_name(),
+                                 candle_interval=time_frame.value,
+                                 candle_closed_only=False,
+                                 symbols=exchange_symbols,
+                                 channels=[cryptofeed_constants.CANDLES],  # pylint: disable=E1101
+                                 callbacks={cryptofeed_constants.CANDLES: candle_callback})  # pylint: disable=E1101
+
+    def subscribe_feeds(self):
+        exchange_symbols = [
+            self.get_exchange_pair(pair)
+            for pair in self.pairs
+        ]
+
+        if self.EXCHANGE_FEEDS.get(Feeds.CANDLE):
+            self.subscribe_candle_feed(exchange_symbols)
+
+        # drop unsupported channels
+        self.channels = [channel for channel in self.channels if channel not in [Feeds.UNSUPPORTED,
+                                                                                 self.EXCHANGE_FEEDS.get(
+                                                                                     Feeds.CANDLE)]]
+        callbacks = {
+            channel: self.callback_by_feed[channel]
+            for channel in self.channels
+        }
+        self.client.add_feed(self.get_feed_name(),
+                             symbols=exchange_symbols,
+                             channels=self.channels,
+                             callbacks=callbacks)
+
+    def start(self):
+        try:
+            self.subscribe_feeds()
+        except Exception as e:
+            self.logger.error(f"Failed to subscribe when creating websocket feed : {e}")
+        try:
+            loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(loop)
```

Comment:
I don't think we need this, cryptofeed [does it already](https://github.com/bmoscon/cryptofeed/blob/master/cryptofeed/feedhandler.py#L247)

Reply:
It was not working when I tried it initialy. I will retest with the current setup